### PR TITLE
feat(sveltekit): upgraded sveltekit support to svelltekit v1.0.0 (fix #392, #397)

### LINKED ISSUE #392: Svelte plugin: incompatibility with svelte-kit starting from 1.0.0-next.574
#### Context

There have been 2 recent releases of svelte-kit that causes `histoire` to fail:

1. [1.0.0-next.574](https://github.com/sveltejs/kit/blob/master/packages/kit/CHANGELOG.md#100-next574), in which the internal `vite-plugin-svelte-kit` was split into `vite-plugin-sveltekit-build` and `vite-plugin-sveltekit-middleware` [in PR #7990](https://github.com/sveltejs/kit/pull/7990).

2. [1.0.0-next.576](https://github.com/sveltejs/kit/blob/master/packages/kit/CHANGELOG.md#100-next576), in which the `sveltekit` vite plugin became `async` ([PR #7994](https://github.com/sveltejs/kit/pull/7994)).

#### Resolution

For (1), please see this [open PR](https://github.com/histoire-dev/histoire/pull/393) that updates `viteIgnorePlugins`.

For (2), I'm not sure there is a way to get around automatically from `@histoire/plugin-svelte` side; I'm happy to help out further if someone can give me some direction. For now, here is a working workaround for me:

```diff
// vite.config.ts/js

import { sveltekit } from '@sveltejs/kit/vite';
+ import { defineConfig } from 'vite';

/** @type {import('vite').UserConfig} */
+ const config = defineConfig(async () => ({
+       // await for for the new async `sveltekit` here
+	plugins: [await sveltekit()]
+ });

export default config;
```

Maybe if nothing can be done, then this should be added to the documentation somewhere?

### Reproduction

Please run `npm run story:dev` from [this StackBlitz](https://stackblitz.com/edit/sveltejs-kit-template-default-ntonhf?file=package.json&terminal=story:dev). Notice:

- verify that the startup site becomes the `svelte-kit` app instead of `histore`,
- if `svelte-kit` is reverted to `1.0.0-next.573`, everything works as expected.

### System Info

```shell
❯ npx envinfo --system --npmPackages '{histoire,@histoire/*,vite,@vitejs/*}' --binaries --browsers
success Install finished in 1.829s


  System:
    OS: Linux 5.0 undefined
    CPU: (4) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
  npmPackages:
    @histoire/controls: 0.11.9 => 0.11.9 
    @histoire/plugin-svelte: 0.11.9 => 0.11.9 
    histoire: 0.11.7 => 0.11.7 
    vite: ^3.2.4 => 3.2.5 

(Ran from the same stackblitz above)
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/histoire-dev/histoire/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/histoire-dev/histoire/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://histoire.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/histoire-dev/histoire/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/histoire-dev/histoire/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by bhvngt**
### Description

This PR makes `histoire` compatible to recent changes in the svelte v1.0.0 They are

- Changes in the vite ignore plugins. see [ref](https://github.com/histoire-dev/histoire/pull/393#discussion_r1045263380) from @benmccann
- Fixed promise resolution of plugin option that was compelling svelte kit users to await sveltekit plugin
- migrated sveltekit example to `v1.0.0`

With above changes, histoire should be able to work with svelte kit v1.0.0 without any changes to their setup.

This PR is a part duplicate of #393, which was addressing the first of the above issues. Shout out to @vnphanquang for initiating this effort.

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [ ] If it's a new feature, provide a convincing reason to add it. *Ideally, you should open a suggestion issue first and have it approved before working on it.*
- [x] Read the [Contributing Guidelines](https://github.com/Akryum/histoire/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/Akryum/histoire/blob/main/CONTRIBUTING.md#pull-request-guidelines) and follow the [Commit Convention](https://github.com/Akryum/histoire/blob/main/.github/commit-convention.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Comment by vnphanquang** [comment]
Great work @bhvngt. I've closed my PR in favor of this. Thanks a lot.

File: packages/histoire/src/node/vite.ts [hunk 1]
```diff
@@ -78,10 +78,11 @@ export async function mergeHistoireViteConfig (viteConfig: InlineConfig, ctx: Co
   let flatPlugins = []
   if (viteConfig.plugins) {
     for (const pluginOption of viteConfig.plugins) {
-      if (Array.isArray(pluginOption)) {
-        flatPlugins.push(...await Promise.all(pluginOption))
+      const resolvedPluginOption = await pluginOption
```
Comment (benmccann) [inline packages/histoire/src/node/vite.ts:81]: I don't understand why this changed. Now if it's an array it's being awaited twice

Comment (benmccann) [reply packages/histoire/src/node/vite.ts:81]: Ok. I guess awaiting twice is actually correct: https://github.com/vitejs/vite/blob/af86e5bcfe4b78f486f499cba09c3270fb151d54/packages/vite/src/node/config.ts#L111

**Review by benmccann — APPROVED** [review]
Thanks!!

File: packages/histoire/src/node/vite.ts (hunk 1 above)
Comment (Akryum) [reply packages/histoire/src/node/vite.ts:81]: A vite plugin can be of the form: `[promise, promise, promise]`

**Comment by benmccann** [comment]
@Akryum anything else we need to do to merge this PR?

**Comment by Akryum** [comment]
@benmccann This PR was pending the merge of the vite 4 branch :wink:

**Pull request merged**